sparse/core.py

```diff
@@ -245,26 +245,43 @@ def __sizeof__(self):
 
     def __getitem__(self, index):
         if not isinstance(index, tuple):
-            index = (index,)
+            if isinstance(index, str):
+                data = self.data[index]
+                idx = np.where(data)
+                coords = []
+
+                for i in range(self.ndim):
+                    coords.append(self.coords[i, idx[0]])
+
+                for i in range(1, np.ndim(data)):
+                    coords.append(idx[i])
+
+                return COO(coords, data.flatten(),
```

Comment:
The `data.flatten()` call seems odd to me here.  This is for struct dtypes?

Reply:
You're right, it should be `data[idx].flatten()`. We have to filter out nonzero entries.